app/controllers/events_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  before_action :require_user, except: [:index]
+
+  def new
+    @users = User.all
+    @users = current_user.except_current_user(@users).pluck(:name)
+    @event = Event.new
+  end
+
+  def show
+    @event = Event.find_by(id: params[:id])
+  end
+
+  def index
```

Comment:
To be able to meet the following requirements we need the past and upcoming event scope as an instance variable:

```
 lists past events, i.e. @past_events = Event.previous_events
 lists ucomping events, i.e. @upcoming_events = Event.upcoming_events
```

Unless you have a specific reason to do it the way you did? It's okay to defend your choice.

[`REQUIRED`]

Reply:
Sorry I am a bit confused about this. I had added this earlier but in odin project instructions they had asked us to use scope instead later which is why I changed it to only using scope and getting it done